### ISSUE TITLE
False positive `SLOT000` on Enum subclasses

### ISSUE THREAD

**Issue by jamesbraza** [astral-sh/ruff#9890]
With `ruff==0.2.1` and Python 3.12:

```python
from enum import Enum

class MyEnum(Enum):
    pass

class CustomStrEnum(str, MyEnum):  # Throws SLOT000
    pass
```

I think this is a false positive for [`SLOT000`](https://docs.astral.sh/ruff/rules/no-slots-in-str-subclass/) based on https://github.com/astral-sh/ruff/issues/5748. At the time, looks like Ruff decided not to check for this.

However, maybe nowadays it's possible, and it is a false positive `SLOT000`. Do you think we can fix this false positive

**Comment by charliermarsh** [comment]
We can fix this for references in the same file.
